common/key-util/src/main/java/io/helidon/common/pki/KeysBuilderDecorator.java

```diff
@@ -93,7 +93,7 @@ private void updateFromKeystore(Keys.BuilderBase<?, ?> builder, KeystoreKeys key
         Optional<String> keyAliasConfigured = keystoreConfig.keyAlias();
         String keyAlias = keyAliasConfigured.orElse(KeystoreKeysBlueprint.DEFAULT_PRIVATE_KEY_ALIAS);
 
-        if (builder.privateKey().isEmpty()) {
+        if (builder.privateKey().isEmpty() || !keystoreConfig.trustStore()) {
```

Comment:
I think this should be `if (builder.privateKey().isEmpty() && !keystoreConfig.trustStore()) {` - i.e. only guess a private key if this is not a trust store and the private key is empty.

Reply:
yes. fixed